Cargo.toml

```diff
@@ -146,7 +147,7 @@ param = ["rustix/param"]
 # [the "personality" feature of the unwinding crate] for more details.
 #
 # [the "personality" feature of the unwinding crate]: https://crates.io/crates/unwinding#personality-and-other-utilities
-eh-personality = ["unwinding?/personality"]
+eh-personality = ["unwinding/personality"]
```

Comment:
Maybe also add an `unwinding` feature enabling the unwinding crate? If you are using libstd, you need to include the unwinding crate to satisfy the dependencies of libstd, but must not use the eh-personality or panic-handler features to prevent conflicts with libstd.

Reply:
I am confused about the precise conditions when one needs to use the unwinding crate versus when an unwinding library is provided automatically. Would it be possible to give a cargo invocation that doesn't work? Then I could add an example-crates configuration that tests that.

Reply:
Also, unwinding is already an optional feature, so you can enable the dependency with --features=unwind, which may be what you're asking for here?

Reply:
> I am confused about the precise conditions when one needs to use the unwinding crate versus when an unwinding library is provided automatically.

An unwinding library should be necessary when either using libstd or using panic=unwind for any crate. With panic=unwind the compiler inserts _Unwind_Resume calls pretty much everywhere, while libstd directly calls into libunwind for getting backtraces and the personality function. If using eyra, the unwinding library needs to be provided by eyra or one of it's dependencies. If not using eyra, it is possible to use the C libgcc or libunwind implementations of an unwinding library provided that you also use the system libc. If you want to use origin with static linking and without libc, the unwinding crate needs to be used.

> Also, unwinding is already an optional feature, so you can enable the dependency with --features=unwind, which may be what you're asking for here?

The rules under which cargo inserts an implicit feature for optional dependencies are not the most clear. If I read the cargo docs correctly the fact that you are using `"unwinding/personality"` here rather than `"dep:unwinding", "unwinding?/personality"` is what causes an implicit feature called `unwinding` to be introduced.

Reply:
A normal cargo build gives you a libunwind automatically that satisfies those `_Unwind_Resume` calls. I am confused specifically about the conditions under which it does or doesn't do that. There are too many dimensions (lto, panic=unwind, panic_immediate_abort, no_std, presence of `#[panic_handler]`, presence of code that actually does unwinding) and every time I think I understand how things work something doesn't work, so I've given up trying and have resorted to just collecting example-crates so that I can frob stuff until all the example-crates work.

My understanding is that all optional dependencies are features unless `dep:` is used, and origin does not use dep:unwinding anywhere, so unwinding is an optional feature.

Reply:
> My understanding is that all optional dependencies are features unless dep: is used

My understanding is that only referencing the crate as dependency of a feature (without `dep:`) or as dependency using `crate_name/feature` makes it an implicit feature with both `dep:crate_name` and `crate_name?/feature` not adding an implicit feature. You are using `unwinding?/feature` in several places.

Reply:
I now have examples in example-crates that use "unwinding"' as a feature, so whatever the rules are, it does appear that "unwinding" is working as a feature.

Reply:
I've now added a feature `unwinding = ["dep:unwinding"]`, to help document that "unwinding" is a feature that users can enable.

Reply:
(... and now with `cfg(panic = "unwind")` in Cargo.toml, it's no longer needed)